test/system/shoot_cp_migration/migrate_test.go

```diff
@@ -0,0 +1,183 @@
+	shootName = flag.String("shoot-name", "", "name of the shoot")
+	shootNamespace = flag.String("shoot-namespace", "", "namespace of the shoot")
+}
+
+var _ = Describe("Shoot migration testing", func() {
+	f := NewGardenerFramework(gardenerConfig)
+	t := NewShootMigrationTest(f, &ShootMigrationConfig{
+		ShootName:      *shootName,
+		ShootNamespace: *shootNamespace,
+		TargetSeedName: *targetSeedName,
+	})
+	guestBookApp := app.GuestBookTest{}
+
+	t.CIt("Before Shoot Migration", func(ctx context.Context) {
+		validateConfig()
+		err := beforeMigration(ctx, t, &guestBookApp)
+
+		ExpectNoError(err)
+	}, CreateAndReconcileTimeout)
+
+	t.CIt("Migrate Shoot", func(ctx context.Context) {
+		err := t.MigrateShoot(ctx)
+		ExpectNoError(err)
+	}, CreateAndReconcileTimeout)
+
+	t.CIt("After Shoot Migration", func(ctx context.Context) {
+		err := afterMigration(ctx, t, guestBookApp)
+		ExpectNoError(err)
+	}, CreateAndReconcileTimeout)
+})
+
+func validateConfig() {
+	if !StringSet(*targetSeedName) {
+		ginkgo.Fail("You should specify a name for the target Seed")
+	}
+	if !StringSet(*shootName) {
+		ginkgo.Fail("You should specify a name for the Shoot that will be migrated")
+	}
+	if !StringSet(*shootNamespace) {
+		ginkgo.Fail("You should specify a namespace of the Shoot that will be migrated")
+	}
+}
+
+func beforeMigration(ctx context.Context, t *ShootMigrationTest, guestBookApp *app.GuestBookTest) error {
+	if err := t.InitShoot(ctx); err != nil {
+		return err
+	}
+	t.SeedShootNamespace = ComputeTechnicalID(t.GardenerFramework.ProjectNamespace, &t.Shoot)
+
+	if err := t.InitSeedsAndClients(ctx); err != nil {
+		return err
+	}
+	t.ComparisonElementsBeforeMigration = ShootComparisonElemets{}
+	if err := t.PopulateBeforeMigrationComparisonElements(ctx); err != nil {
+		return err
+	}
+	if err := t.CreateTestSecret(ctx); err != nil {
+		return err
+	}
+	if err := t.CreateTestServiceAccount(ctx); err != nil {
+		return err
+	}
+	initializedApp, err := initGuestBookTest(ctx, t)
+	if err != nil {
+		return err
+	}
+	*guestBookApp = *initializedApp
+	guestBookApp.DeployGuestBookApp(ctx)
+	guestBookApp.Test(ctx)
+	podStatusMap, err := t.GetPodsStatus(ctx)
+	if err != nil {
+		return err
+	}
+	t.ComparisonElementsBeforeMigration.PodStatus = podStatusMap
+	return nil
+}
+
+func afterMigration(ctx context.Context, t *ShootMigrationTest, guestBookApp app.GuestBookTest) error {
+	if ginkgo.CurrentGinkgoTestDescription().Failed {
+		t.GardenerFramework.DumpState(ctx)
+	}
+	if err := t.PopulateAfterMigrationComparisonElements(ctx); err != nil {
+		return err
+	}
+
+	if err := t.DeleteTestSecret(ctx); err != nil {
+		return err
+	}
+	if err := t.DeleteTestServiceAccount(ctx); err != nil {
+		return err
+	}
+	podStatusMap, err := t.GetPodsStatus(ctx)
+	if err != nil {
+		return err
+	}
+	t.ComparisonElementsAfterMigration.PodStatus = podStatusMap
+	if err := t.CompareElementsAfterMigration(); err != nil {
+		return err
+	}
+	fmt.Printf("#### %v\n", guestBookApp)
```

Comment:
Shouldn't we use logger instead? Or it is just a leftover from debugging?

Reply:
It was a leftover